test/integration/puppetlabs/services/jruby_pool_manager/jruby_pool_int_test.clj

```diff
@@ -376,7 +300,7 @@
 
          (jruby-core/flush-pool! pool-context)
          ; wait until the flush is complete
-         (await (jruby-agents/get-pool-agent pool-context))
+         (is (jruby-testutils/wait-for-instances (jruby-internal/get-pool pool-context) 1))
```

Comment:
I think we had the `await` in there because it should block until the flush action has been completely processed through the agent.  We knew that on returning from the `flush-pool!` call that the flush action had at least been sent to the agent.  Under the hood, I think the `await` sends another action to the agent and waits for that to be processed.  By the time the second one has gone through, the first one should have finished.  Seems like this should still be the case now for determining when the flush has been completed on the pool agent?

> I think we were talking about how making the flush completely synchronous might be a good idea.

From the perspective of the HTTP flush jruby-pool endpoint, I think I'm okay with just the processing down through `collect-all-jrubies` being guaranteed to complete but not the actual refilling of the pool.  Waiting for the pool to completely fill back up would make the HTTP calls last longer.  Seems good enough to me if the caller can at least be guaranteed that any subsequent use of JRubies from the pool will be against new ones.